### PR TITLE
Find tinyxml

### PR DESCRIPTION
**Pull request by kpp**
Use FindTinyXML to find package
Implements
```
 libfind_version_n_header(<PREFIX>
  NAMES <name> [<name> ...]
  DEFINES <define> [<define> ...] | CONSTANTS <const> [<const> ...]
  [PATHS <path> [<path> ...]]
  [QUIET]
)
```

Usage:
```
libfind_version_n_header(TinyXML
    NAMES tinyxml.h
    CONSTANTS TIXML_MAJOR_VERSION TIXML_MINOR_VERSION TIXML_PATCH_VERSION
)
libfind_version_n_header(Freetype
    NAMES freetype/freetype.h freetype.h
    PATHS Freetype_OLD_INCLUDE_DIR
    DEFINES FREETYPE_MAJOR FREETYPE_MINOR FREETYPE_PATCH
)
```

### CODE REVIEW

File: extern/oics/CMakeLists.txt [hunk 1]
```diff
@@ -1,32 +1,22 @@
...
-set(TINYXML_SOURCE_FILES
-	tinyxml.cpp
-	tinyxmlparser.cpp
-	tinyxmlerror.cpp
-	tinystr.cpp	
+add_library(local_tinyxml STATIC
+    tinyxml.cpp
+    tinyxmlparser.cpp
+    tinyxmlerror.cpp
+    tinystr.cpp
```
Comment (scrawl) [inline extern/oics/CMakeLists.txt]: We don't need to build these if USE_SYSTEM_TINYXML is true.

Comment (kpp) [inline extern/oics/CMakeLists.txt]: Done

**Comment by kpp** [comment]
@scrawl hello?..

File: CMakeLists.txt [hunk 2]
```diff
@@ -676,8 +669,6 @@ if (WIN32)
     endforeach(d)
 
     set_target_properties(components PROPERTIES COMPILE_FLAGS "${WARNINGS} ${MT_BUILD}")
-    # oics uses tinyxml, which has an initialized but unused variable
```
Comment (kpp) [inline CMakeLists.txt]: This one was suppressed due to 
> include_directories(**SYSTEM** ${TinyXML_INCLUDE_DIRS})

File: extern/oics/CMakeLists.txt [hunk 3]
```diff
@@ -1,32 +1,21 @@
...
+    add_library(local_tinyxml STATIC
+        tinyxml.cpp
+        tinyxmlparser.cpp
+        tinyxmlerror.cpp
+        tinystr.cpp
+    )
+    target_link_libraries(oics local_tinyxml)
 endif()
-
-# Does this do anything?
```
Comment (kpp) [inline extern/oics/CMakeLists.txt]: No

File: cmake/LibFindMacros.cmake [hunk 4]
```diff
@@ -54,43 +56,138 @@ function (libfind_pkg_detect PREFIX)
...
+  # Export the version string
+  string(REPLACE ";" "." version "${version_parts}")
+  set(${PREFIX}_VERSION "${version}" PARENT_SCOPE)
+endfunction()
+
+# libfind_version_header(<PREFIX> <HEADER> <DEFINE_NAME> [PATHS <path> [<path> ...]] [QUIET])
+# Extracts a version #define from a version.h file, output stored to <PREFIX>_VERSION.
+# This function does nothing if the version variable is already defined.
+# Usage: libfind_version_header(Foobar foobar/version.h FOOBAR_VERSION_STR)
+function (libfind_version_header PREFIX VERSION_H DEFINE_NAME)
```
Comment (kpp) [inline cmake/LibFindMacros.cmake]: Rewritten with call to `libfind_version_n_header` with one name

**Comment by scrawl** [comment]
Sorry, I forgot about this PR.

**Pull request merged**